File: src/optic/lua/ssa/instructions/Assign.java
package optic.lua.ssa.instructions;

import optic.lua.ssa.*;

import java.util.List;

public final class Assign implements Step {
	private final List<Register> sources;
	private final List<LValue> targets;

	public Assign(List<LValue> targets, List<Register> sources) {
		this.targets = targets;
		this.sources = sources;
	}

	@Override
	public String toString() {
		return "assign " + targets + " = " + sources;
	}
}

File: src/optic/lua/ssa/instructions/Assign.kt
package optic.lua.ssa.instructions

import optic.lua.ssa.LValue
import optic.lua.ssa.Register
import optic.lua.ssa.Step

class Assign(val targets: List<LValue>, val sources: List<Register>) : Step {
    override fun toString(): String {
        return "assign $targets = $sources"
    }
}
